Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/CoordinatedGraphicsScene.cpp

```diff
@@ -81,29 +67,28 @@ void CoordinatedGraphicsScene::paintToCurrentGLContext(const TransformationMatri
 
     bool sceneHasRunningAnimations = currentRootLayer->applyAnimationsRecursively(MonotonicTime::now());
 
+    bool didChangeClipRect = false;
     FloatRoundedRect actualClipRect(clipRect);
-#if ENABLE(WPE_PLATFORM) || PLATFORM(GTK)
-    if (m_propagateDamage != Damage::ShouldPropagate::No) {
+#if ENABLE(DAMAGE_TRACKING)
+    if (m_client && !sceneHasRunningAnimations && m_damagePropagation != Damage::Propagation::None) {
+        Damage damage;
         WTFBeginSignpost(this, CollectDamage);
-        currentRootLayer->collectDamage(*m_textureMapper);
+        currentRootLayer->collectDamage(*m_textureMapper, damage);
         WTFEndSignpost(this, CollectDamage);
 
-        WebCore::Damage boundsDamage;
-        const auto& frameDamage = ([this, &boundsDamage, &unifyDamagedRegions]() -> const WebCore::Damage& {
-            const auto& damage = lastDamage();
-            if (m_propagateDamage != Damage::ShouldPropagate::No && !damage.isInvalid()) {
-                if (unifyDamagedRegions) {
-                    boundsDamage.add(damage.bounds());
-                    return boundsDamage;
-                }
-                return damage;
+        Damage boundsDamage;
+        const auto& frameDamage = ([this, &damage, &boundsDamage]() -> const Damage& {
+            if (m_damagePropagation == Damage::Propagation::Unified && !damage.isInvalid()) {
+                boundsDamage.add(damage.bounds());
+                return boundsDamage;
             }
-            return WebCore::Damage::invalid();
+            return damage;
         })();
-        if (m_client) {
-            const Damage& damageSinceLastSurfaceUse = m_client->addSurfaceDamage(frameDamage);
-            if (!damageSinceLastSurfaceUse.isInvalid())
-                actualClipRect = static_cast<FloatRoundedRect>(damageSinceLastSurfaceUse.bounds());
+
+        const auto& damageSinceLastSurfaceUse = m_client->addSurfaceDamage(frameDamage);
```

Comment:
this should be happening regardless of `sceneHasRunningAnimations` as otherwise it may not be called some frames and we may end up with wrong per-RenderTarget-damage

Reply:
Fixed